app/assets/javascripts/metrics.js

```diff
@@ -1,56 +1,222 @@
+					break;
+				default:
+					console.error(condition?.condition_type +" is not valid");
+					return;
 			}
-		});
+		})
+	}
+
+	$('.ui.form').change(function() {
+		$('#save').removeClass('disabled');
+	});
+
+	$(window).bind('beforeunload', function(){
+		if (!$('#save').hasClass('disabled')) {
+			return 'Make sure to save your changes';
+		}
+	});
 });
+
+$('#save').click(function(){
+
+	// Checks form validity
+	if(!$('.ui.form').form('is valid'))
+		return;
+	
+	$('#save').addClass('loading');
+	let new_conditions = {};
+
+	if($('#grace_days_checkbox').checkbox('is checked')){
+		new_conditions['grace_day_usage'] = {
+			grace_day_threshold: $("#grace_days_value").dropdown('get value'),
+			date: $('#grace_days_by_date').calendar('get date')
+		};
+	}
+
+	if($('#grade_drop_checkbox').checkbox('is checked')){
+		new_conditions['grade_drop'] = {
+			percentage_drop: $("#grade_drop_percentage").val(),
+			consecutive_counts: $('#grade_drop_consecutive_counts').dropdown('get value')
+		};
+	}
+
+	if($('#no_submit_checkbox').checkbox('is checked')){
+		new_conditions['no_submissions'] = {
+			no_submissions_threshold: $("#no_submit_value").dropdown('get value')
+		};
+	}
+
+	if($('#low_grades_checkbox').checkbox('is checked')){
+		new_conditions['low_grades'] = {
+			grade_threshold: $("#low_grades_percentage").val(),
+			count_threshold: $('#low_grades_count').dropdown('get value')
+		};
+	}
+
+	$.ajax({
+		url:metrics_endpoints['update'],
+		dataType: "json",
+		contentType:'application/json',
+		data: JSON.stringify(new_conditions),
+		type: "POST",
+		success:function(data){
+			render_banner({
+				type:"positive",
+				header:"You have successfully saved your conditions",
+				message:"Your watchlist should reflect your new conditions"
+			});
+		},
+		error:function(result, type){
+			render_banner({
+				type:"negative",
+				header:"Currently unable to update conditions",
+				message: "Do try again later",
+				timeout: -1
+			});
+		},
+		complete:function(){
+			$('#save').removeClass('loading');
+			$('#save').addClass('disabled');
+		}
+	});
+})
+
+// variable to keep track of the different banners
+var message_count = 0;
+
+/**
+ * Renders a banner given parameters
+ * @param {Object} params Parameters of the banner
+ * @param {string} params.type type of the banner, positive, negative, or warning
+ * @param {string} params.header html string header of the banner
+ * @param {string} params.message html string body of the banner
+ * @param {number} params.timeout timeout of banner in milleseconds. -1 for no timeout. 
+ */
+const render_banner = (params) => {
+	
+	const message_id = message_count; // using a count as id
+
+	// templating the html
+	// important to give each element an unique id
+	const message_html = `
+	<div class="ui ${params.type} message" id="message_${message_id}">
+		<i class="close icon"></i>
+		<div class="header">
+		${params.header}
+		</div>
+		${params.message}
+	</div>
+	`;
```

Comment:
the main useful thing to utilize in creating HTML elements is template strings (https://hacks.mozilla.org/2015/05/es6-in-depth-template-strings-2/)

It's not supported by I.E at all, but this is just so clean. In fact you can create another function, that generates another HTML string using the template string, and then insert it here. (In this case, imagine params.message is <div><p>hello</p></div> generated using another function)